### PR TITLE
CI/TYP: set python_version for mypy

### LINKED ISSUE #43807: TYP: mypy complaint isocalendar does not match supertype
### 

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
When I run mypy locally (this has happened for a while and I've been ignoring it) I get


pandas/_libs/tslibs/timestamps.pyi:160: error: Return type "Tuple[int, int, int]" of "isocalendar" incompatible with return type "_IsoCalendarDate" in supertype "datetime"  [override]
pandas/_libs/tslibs/timestamps.pyi:160: error: Return type "Tuple[int, int, int]" of "isocalendar" incompatible with return type "_IsoCalendarDate" in supertype "date"  [override]
```
```


### Issue Description

N/A

### Expected Behavior

N/A

### Installed Versions

py39, master

### PR DESCRIPTION
**Pull request by twoertwein**
- [x] closes #43807
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
cc @simonjayhawkins

File: pyproject.toml [hunk 1]
```diff
@@ -69,6 +69,7 @@ markers = [
 ]
 
 [tool.mypy]
+python_version = "3.8"
```
Comment (simonjayhawkins) [inline pyproject.toml:72]: can you move this adjacent to `platform = "linux-64"`

The config is ordered the same as the mypy docs.

Comment (twoertwein) [reply pyproject.toml]: moved

**Review by simonjayhawkins — APPROVED** [review]
Thanks @twoertwein lgtm. merge on green.

**Comment by jreback** [comment]
thanks @twoertwein do we need to mention this in the docs?

**Pull request merged**